### PR TITLE
Feat/unit conversion batch api

### PR DESCRIPTION
**Pull request by danieljacquin**
Endpoint:
- Add BatchItem, BatchRequest, BatchResponse types
- Add ConvertBatch service method with per-operation error handling
- Add POST /convert/batch endpoint via httpx.HandleBatch
- Use *float64 for Value to distinguish absent field from zero

Tests:
- Add service tests for batch conversion and preserved order
- Add transport tests for happy path, empty operations, limit exceeded
- Add transport test for X-Usage-Count response header

Docs:
- Add documentation for POST /convert/batch
- update endpoints_count from 2 to 3 

## Summary by CodeRabbit

* **New Features**
  * Added batch unit conversion (POST /v1/technology/convert/batch) supporting up to 50 operations, returning per-item results with success/error flags, conversion data, and overall total
  * Response includes a usage-count header

* **Bug Fixes / Validation**
  * Empty or oversized operation lists return validation errors (422)
  * Items missing a value are marked unsuccessful with an error ("value is required")

* **Documentation**
  * Added API docs with multi-language examples

* **Tests**
  * Added coverage for batch endpoint behaviors and validations

[Review Change Stack](https://app.coderabbit.ai/change-stack/bobadilla-tech/requiems-api/pull/637?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Review by crydafan — CHANGES_REQUESTED** [review]
Don't forget to check the comments left by CodeRabbit

File: apps/api/services/technology/units/service.go [hunk 1]
```diff
@@ -225,3 +226,36 @@ func formatFactor(f float64) string {
...
+
+	for i, op := range operations {
+		result, err := s.Convert(op.From, op.To, *op.Value)
+		if err != nil {
+			results[i] = BatchResponse{
+				From:    op.From,
+				To:      op.To,
+				Success: false,
+				Error:   err.Error(),
+			}
```
[the remark this answers is not in the document]
Comment (danieljacquin) [reply apps/api/services/technology/units/service.go:254]: Value is validated as required at the transport layer before ConvertBatch is ever called. A nil Value will result in a 422 Unprocessable Entity response and never reach this code path. The nil check would be dead code here.

[the remark this answers is not in the document]
Comment (crydafan) [reply apps/api/services/technology/units/service.go:254]: @danieljacquin even if the transport layer enforces the validation we should add those guards inside the service functions to be more verbose on the behavior of the functions, CodeRabbit is right to mention tests. Check what negative testing is and implement a test case where the payload for the service function is malformed and it should return an empty result.

**Review by crydafan — APPROVED** [review]
Nicely done Daniel!
